src/main/MainScreen.js

```diff
@@ -22,12 +22,28 @@ export default class MainScreen extends PureComponent<Props> {
     this.props.navigation.navigate('DrawerOpen');
   };
 
+  constructor(props, context) {
+    super(props, context);
+    this.state = { shouldShowStatusBar: (
+      Dimensions.get('window').width < Dimensions.get('window').height
+    ) };
+  }
+
+  reloadOrientation() {
+    this.setState({ shouldShowStatusBar: (
+      Dimensions.get('window').width < Dimensions.get('window').height
+    ) });
+  }
+
   render() {
     const { styles } = this.context;
 
     return (
-      <View style={[styles.flexed, styles.backgroundColor]}>
-        <ZulipStatusBar />
+      <View
+        onLayout={() => { this.reloadOrientation(); }}
+        style={[styles.flexed, styles.backgroundColor]}
+      >
+        { this.state.shouldShowStatusBar ? <ZulipStatusBar /> : null }
```

Comment:
just check here from `orientation`

Reply:
And we can further simply it to `{orientation === 'portrait' && <ZulipStatusBar />}` (no `null` required)
like https://github.com/zulip/zulip-mobile/blob/master/src/chat/Chat.js#L67